README.md

```diff
@@ -1,20 +1,40 @@
 # Demo
 
-To start your Phoenix server:
+## Spec
 
-  * Install dependencies with `mix deps.get`
-  * Create and migrate your database with `mix ecto.setup`
-  * Install Node.js dependencies with `npm install --prefix assets`
-  * Start Phoenix endpoint with `mix phx.server`
+We will have two database tables: "users" and "users_tokens":
 
-Now you can visit [`localhost:4000`](http://localhost:4000) from your browser.
+* "users" will have the "hashed_password", "email" and "confirmed_at" fields plus timestamps
+* "users_tokens" will have "token", "context", "sent_to", "inserted_at" fields
```

Comment:
Have you considered moving away from pre-required/specific database tables?

Consider the scenario where the _user_ table already exists, so all they need to do is to add the missing columns and use your logic. This would virtually enable absolute flexibility and stay out of the end-user/dev way. For example, the way I look at this current implementation is that the spec assumes there will be a password always (which is not true).

For example, you could consider replacing the `users_tokens` with something like JWT. Where you sign (or even encrypt) the token payload. You decrypt/verify it in your logic instead of bringing the DB as a requirement.

I recently worked on [a passwordless authentication solution](https://github.com/stas/otp-jwt/), and it's unfortunate that most (if not all) the existing authentication systems are strictly following patterns that will always stay in your developer way and most probably enforce some pretty painful UX on the end-user side.

Reply:
I personally have no interest in covering all possible use cases. :) I have done that, it didn't work. I intend this solution to be pretty much as is. If you want to move away from what it does, you can either:

1. Use this generator and then modify the generated result accordingly
2. Use this generator as a template for your own generator
3. Don't use this generator at all and roll your own (or use a lib)

In regards to JWT and in general non DB-backed tokens, we discussed this extensively, and we believe that DB tokens are more secure and provide better UI, as they allow you to expire and track them individually.

In any case, I don't guarantee this generator provides the best UI possible, if you believe that passwordless is the best way to go for certain apps, then I look forward to a `mix phx.gen.passless` - you certainly already did the hardest part of the work. :)

Reply:
Thanks for explaining this @josevalim 

I arrived here having in mind Devise as a context. I'm glad this is a more specific solution vs. all-case one.

Reply:
Oh, in case you didn't see it, there is more context here: https://dashbit.co/blog/a-new-authentication-solution-for-phoenix

I was not expecting folks to find this PR outside of the blog post. :)